### PR TITLE
Adding an MCP backend to Mixer [Galley Integration]

### PR DESCRIPTION
**Pull request by ozevren**
This PR introduces an MCP-based backend to Mixer, that can perform demultiplexing of legacy Mixer resources.

### CODE REVIEW

File: mixer/pkg/config/mcp/backend.go [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+	"istio.io/istio/galley/pkg/kube/converter/legacy"
+
+	mcp "istio.io/api/mcp/v1alpha1"
+	"istio.io/istio/galley/pkg/mcp/client"
+	"istio.io/istio/mixer/pkg/config/store"
+	"istio.io/istio/pkg/log"
+	"istio.io/istio/pkg/probe"
+)
+
+var scope = log.RegisterScope("mcp", "Mixer MCP client stack", 0)
```
Comment (geeknoid) [inline mixer/pkg/config/mcp/backend.go]: The MCP library itself doesn't add a scope, does it?

File: mixer/pkg/config/mcp/conversion.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+	urlToKind := make(map[string]string)
+	for _, spec := range schema.All() {
+		if _, ok := mixerKinds[spec.Kind]; ok {
+			kindToURL[spec.Kind] = spec.Target.TypeURL.String()
+			urlToKind[spec.Target.TypeURL.String()] = spec.Kind
+		}
+	}
+
+	if len(mixerKinds) != len(kindToURL) {
+		// We couldn't metadata for some of the well-known Mixer kinds. This shouldn't happen
```
Comment (geeknoid) [inline mixer/pkg/config/mcp/conversion.go]: couldn't find

File: mixer/pkg/config/mcp/conversion.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+		Namespace: ns,
+		Name:      localName,
+	}
+}
+
+func toBackendResource(key store.Key, resource proto.Message, version string) (*store.BackEndResource, error) {
+	// TODO: Ensure that this conversion is correct.
+	marshaller := jsonpb.Marshaler{}
+	jsonData, err := marshaller.MarshalToString(resource)
+	//jsonData, err := json.Marshal(resource)
```
Comment (geeknoid) [inline mixer/pkg/config/mcp/conversion.go]: Forgotten code?

File: mixer/pkg/runtime/runtime.go [hunk 4]
```diff
@@ -159,14 +159,14 @@ func (c *Runtime) processNewConfig() {
 
 	builder := compiled.NewBuilder(newSnapshot.Attributes)
 	newRoutes := routing.BuildTable(
-		newHandlers, newSnapshot, builder, c.defaultConfigNamespace, log.DebugEnabled())
+		newHandlers, newSnapshot, builder, c.defaultConfigNamespace, true)
```
Comment (geeknoid) [inline mixer/pkg/runtime/runtime.go]: Do we really want this?

File: mixer/pkg/runtime/runtime.go [hunk 5]
```diff
@@ -159,14 +159,14 @@ func (c *Runtime) processNewConfig() {
...
 
 	oldContext := c.dispatcher.ChangeRoute(newRoutes)
 
 	c.handlers = newHandlers
 	c.snapshot = newSnapshot
 
-	log.Debugf("New routes in effect:\n%s", newRoutes)
+	log.Infof("New routes in effect:\n%s", newRoutes)
```
Comment (geeknoid) [inline mixer/pkg/runtime/runtime.go]: I don't think this is relevant for our users, debug is better.

File: mixer/pkg/config/mcp/backend.go (hunk 1 above)
Comment (ozevren) [reply mixer/pkg/config/mcp/backend.go]: No, the client library doesn't. (There is one in server though, but it doesn't affect Mixer).

File: mixer/pkg/config/mcp/conversion.go (hunk 3 above)
Comment (ozevren) [reply mixer/pkg/config/mcp/conversion.go]: embarassing.

File: mixer/pkg/runtime/runtime.go (hunk 4 above)
Comment (ozevren) [reply mixer/pkg/runtime/runtime.go]: Sorry. This is highly useful when debugging, but I meant to revert it before checking in.

**Pull request merged**